### PR TITLE
Handle empty body messages using mail v0.2.1

### PR DESCRIPTION
**Pull request by hugomaiavieira**
Version 0.2.1 has [introduced a change](https://github.com/DockYard/elixir-mail/commit/89d05a88b5e8ab73a4a48c829356d98878a08b39) that doesn't allow `nil` as body massages anymore.

### CODE REVIEW

**Comment by hugomaiavieira** [comment]
Also, what do you think about the refactoring bellow?

<details><summary>click to expand the diff</summary>
<p>

```diff
diff --git a/lib/bamboo/adapters/ses_adapter.ex b/lib/bamboo/adapters/ses_adapter.ex
index eda35f3..295020f 100644
--- a/lib/bamboo/adapters/ses_adapter.ex
+++ b/lib/bamboo/adapters/ses_adapter.ex
@@ -7,7 +7,6 @@ defmodule Bamboo.SesAdapter do
 
   @behaviour Bamboo.Adapter
 
-  alias Bamboo.Attachment
   alias Bamboo.SesAdapter.RFC2822WithBcc
   alias ExAws.SES
   import Bamboo.ApiError
@@ -20,39 +19,37 @@ defmodule Bamboo.SesAdapter do
     config
   end
 
-  def deliver(email, _config) do
-    message =
-      Mail.build_multipart()
-      |> Mail.put_from(prepare_address(email.from))
-      |> Mail.put_reply_to(email.headers["Reply-To"])
-      |> Mail.put_to(prepare_addresses(email.to))
-      |> Mail.put_cc(prepare_addresses(email.cc))
-      |> Mail.put_bcc(prepare_addresses(email.bcc))
-      |> Mail.put_subject(email.subject)
-      |> put_text(email.text_body)
-      |> put_html(email.html_body)
-    message = email.attachments
-              |> Enum.map(&prepare_file(&1))
-              |> Enum.reduce(message, &Mail.put_attachment(&2, &1))
-    raw_message = Mail.render(message, RFC2822WithBcc)
-
-    email = SES.send_raw_email(raw_message)
-    case email |> ExAws.request do
+  def deliver(email, config) do
+    ex_aws_config = Map.get(config, :ex_aws, [])
+
+    case Mail.build_multipart()
+         |> Mail.put_from(prepare_address(email.from))
+         |> Mail.put_reply_to(email.headers["Reply-To"])
+         |> Mail.put_to(prepare_addresses(email.to))
+         |> Mail.put_cc(prepare_addresses(email.cc))
+         |> Mail.put_bcc(prepare_addresses(email.bcc))
+         |> Mail.put_subject(email.subject)
+         |> put_text(email.text_body)
+         |> put_html(email.html_body)
+         |> put_attachments(email.attachments)
+         |> Mail.render(RFC2822WithBcc)
+         |> SES.send_raw_email()
+         |> ExAws.request(ex_aws_config) do
       {:ok, response} -> response
       {:error, reason} -> raise_api_error(inspect(reason))
     end
   end
 
-  defp prepare_file(%Attachment{} = attachment) do
-    {attachment.filename, attachment.data}
+  def put_attachments(message, []), do: message
+
+  def put_attachments(message, attachments) do
+    Enum.reduce(attachments, message, &Mail.put_attachment(&2, {&1.filename, &1.data}))
   end
 
   def put_text(message, nil), do: message
-
   def put_text(message, body), do: Mail.put_text(message, body)
 
   def put_html(message, nil), do: message
-
   def put_html(message, body), do: Mail.put_html(message, body)
 
   defp prepare_addresses(recipients) do
```

</p>
</details>

**Comment by kalys** [comment]
@hugomaiavieira 
> Also, what do you think about the refactoring bellow?

Looks great. Could you create a PR?

**Comment by kalys** [comment]
@hugomaiavieira Also could you resolve conflict?

**Comment by hugomaiavieira** [comment]
@kalys done.

File: test/lib/bamboo/adapters/ses_adapter_test.exs [hunk 1]
```diff
@@ -55,6 +55,23 @@ defmodule Bamboo.SesAdapterTest do
     new_email() |> SesAdapter.deliver(%{})
   end
 
+  test "delivers successfully email withtout body" do
```
[the remark this answers is not in the document]
Comment (hugomaiavieira) [reply test/lib/bamboo/adapters/ses_adapter_test.exs]: @kalys done

**Pull request merged**